src/components/fields/date-field/date-field.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import requiredIf from 'react-required-if';
+import Constraints from '../../constraints';
+import Spacings from '../../spacings';
+import FieldLabel from '../../field-label';
+import DateInput from '../../inputs/date-input';
+import createSequentialId from '../../../utils/create-sequential-id';
+import FieldErrors from '../../field-errors';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+
+const sequentialId = createSequentialId('date-field-');
+
+const hasErrors = errors => errors && Object.values(errors).some(Boolean);
+
+class DateField extends React.Component {
+  static displayName = 'DateField';
+
+  static isEmpty = DateInput.isEmpty;
+
+  static propTypes = {
+    // DateField
+    id: PropTypes.string,
+    horizontalConstraint: PropTypes.oneOf(['xs', 's', 'm', 'l', 'xl', 'scale']),
+    errors: PropTypes.shape({
+      missing: PropTypes.bool,
+    }),
+    renderError: PropTypes.func,
+    isRequired: PropTypes.bool,
+    touched: PropTypes.bool,
+
+    // DateInput
+    name: PropTypes.string,
+    value: PropTypes.string.isRequired,
+    onChange: requiredIf(PropTypes.func, props => !props.isReadOnly),
+    onBlur: PropTypes.func,
+    onFocus: PropTypes.func,
+    isDisabled: PropTypes.bool,
+    placeholder: PropTypes.string,
+
+    // LabelField
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+    hint: requiredIf(
+      PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+      props => props.hintIcon
+    ),
+    description: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    onInfoButtonClick: PropTypes.func,
+    hintIcon: PropTypes.node,
+    badge: PropTypes.node,
+  };
+
+  static defaultProps = {
+    horizontalConstraint: 'scale',
+  };
+
+  state = {
+    // We generate an id in case no id is provided by the parent to attach the
+    // label to the input component.
+    id: this.props.id,
+  };
+
+  static getDerivedStateFromProps = (props, state) => ({
+    id: do {
```

Comment:
This is a nice use case for a HoC or hook.

Reply:
Yep, already on that :P I don't think it could be a hook, but I'm working on a HoC for it. See #308